src/permutation.rs

```diff
@@ -77,30 +77,50 @@ macro_rules! permutation_trait(
 
 macro_rules! permutation_trait_impl(
   ($($name:ident $ty:ident $item:ident),+) => (
-    impl<
-      Input: Clone, $($ty),+ , Error: ParseError<Input>,
-      $($name: Parser<Input, $ty, Error>),+
-    > Permutation<Input, ( $($ty),+ ), Error> for ( $($name),+ ) {
+    impl<'a,
+      $($ty),+ , Error: std::fmt::Debug + ParseError<&'a[u8]>,
+      $($name: Parser<&'a [u8], $ty, Error>),+
+    > Permutation<'a, ( $(Option<$ty>),+ ), Error> for ( $($name),+ ) {
+
+      fn permutation(&mut self, mut input: &'a [u8]) -> IResult<&'a [u8], ( $(Option<$ty>),+ ), Error> {
+        let i_ = input.clone();
 
-      fn permutation(&mut self, mut input: Input) -> IResult<Input, ( $($ty),+ ), Error> {
         let mut res = ($(Option::<$ty>::None),+);
 
-        loop {
+        // Sum of the permutation errors obtained applying each parser
+        let permutation_error = loop {
           let mut err: Option<Error> = None;
+
+          // Skip void elements
+          let void_res = crate::ebml::skip_void(input);
```

Comment:
It's not just `void` that you want to skip parsing but also all ebml elements that are not handled

Reply:
You're right, but I wanted to add that later, in another PR. This PR was meant to fix the current tests. Indeed I haven't mentioned the issue about the not handled ebml elements

Reply:
Yeah, I think that will be the solution, but we can do that in a later PR so we don't mix up concerns :)

Reply:
Sure, works for me :)